kong/plugins/key-auth/handler.lua

```diff
@@ -17,13 +17,20 @@ local KeyAuthHandler = BasePlugin:extend()
 
 KeyAuthHandler.PRIORITY = 1000
 
+local OPTIONS = "OPTIONS"
+
 function KeyAuthHandler:new()
   KeyAuthHandler.super.new(self, "key-auth")
 end
 
 function KeyAuthHandler:access(conf)
   KeyAuthHandler.super.access(self)
 
+  -- do not authenticate preflight OPTIONS requests
+  if ngx.req.get_method() == OPTIONS then
+    return
+  end
+
```

Comment:
I'm not an expert on this, but this completely disables the key-auth for `options` requests. Shouldn't this at least be configurable?